src/main/java/org/jvnet/hudson/test/JenkinsRule.java

```diff
@@ -671,7 +708,34 @@ public File getWebAppRoot() throws Exception {
      * that we need for testing.
      */
     protected ServletContext createWebServer() throws Exception {
-        server = new Server(new ThreadPoolImpl(new ThreadPoolExecutor(10, 10, 10L, TimeUnit.SECONDS, new LinkedBlockingQueue<Runnable>(),new ThreadFactory() {
+        ImmutablePair<Server,  ServletContext> results = _createWebServer(contextPath, (x) -> localPort = x, () -> {
+            try {
+                return getURL();
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "Unable to get Url", e);
+                return null;
```

Comment:
And what is returning null going to do? Something else will break. Suggest replacing both lines with a simple

```java
throw new AssertionError(e);
```

unless this closure already permits `IOException` to be thrown, in which case delete the whole `try` block.

Reply:
I've removed it entirely